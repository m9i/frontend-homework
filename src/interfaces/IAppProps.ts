export interface IAppProps {
    showDropdown: any;
    hideDropdown: any;
}
