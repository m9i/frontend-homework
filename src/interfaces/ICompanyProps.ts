import {ICompany} from './ICompany';

export interface ICompanyProps {
    onCompanyChange: any;
    selectedCompany: ICompany;
    company: ICompany;
    hideDropdown: any;
}
