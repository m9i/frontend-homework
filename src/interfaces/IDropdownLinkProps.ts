import {IToggleDropdownAction} from '../store/actions/toggleDropdownAction';

export interface IDropdownLinkProps {
    toggle: boolean;
    showDropdown: (() => (IToggleDropdownAction));
    hideDropdown: (() => (IToggleDropdownAction));
    selectedCompany: any;
}
