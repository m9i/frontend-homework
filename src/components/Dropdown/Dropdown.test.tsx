import React from 'react';
import DropDown from './Dropdown';
import {mount, ReactWrapper, shallow} from 'enzyme';
import {Provider} from 'react-redux';
import {ICompany} from '../../interfaces/ICompany';
import configureMockStore from 'redux-mock-store';

describe('DropDown', () => {
    const mockStore = configureMockStore([]);
    let store: any;
    let companyInfo: ICompany;

    beforeEach(() => {
        companyInfo = {id: 123, value: 'mock-name'};
        store = mockStore({
            dropdown: {},
            company: {
                companies: [
                    {id: 123, value: 'mock-company-1'},
                    {id: 124, value: 'mock-company-1'}
                ],
                selectedCompany: {id: 123, value: 'mock-company-1'}
            },
        });
    });


    it('snapshot', () => {
        const DropDownComponent = shallow(
            <Provider store={store}>
                <DropDown/>
            </Provider>
        ).dive();

        expect(DropDownComponent).toMatchSnapshot();
    });

    it('should render companies list', () => {
        const DropDownComponent = mount(
            <Provider store={store}>
                <DropDown/>
            </Provider>
        );

        const list: ReactWrapper = DropDownComponent.find('Company');
        expect(list.length).toBe(2);
    });
});
