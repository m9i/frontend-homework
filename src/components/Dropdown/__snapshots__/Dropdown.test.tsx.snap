// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown snapshot 1`] = `<Connect(Dropdown) />`;
